sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -145,7 +146,7 @@ class RapidsBufferCatalog extends Logging {
   def removeBuffer(id: RapidsBufferId): Unit = {
     val buffers = bufferMap.remove(id)
     if (buffers != null) {
```

Comment:
do we need this null check? It seems the implicit checks that the sequence is not null, so we should be really safe to just do:
```
val buffers = bufferMap.remove(id)
buffers.safeFree()
```